velox/common/base/IOUtils.h

```diff
@@ -18,7 +18,16 @@
 #include <cstdint>
 #include <cstring>
 
+#include <folly/io/IOBuf.h>
+
+namespace facebook::velox {
+struct ByteRange;
+}
+
 namespace facebook::velox::common {
+
+std::vector<ByteRange> writableByteRangesFromIOBuf(folly::IOBuf* iobuf);
```

Comment:
Maybe just call it byteRangesFromIOBuf. It is used for input stream but byte range only take non-const pointer.